include/cutlass/cutlass.h

```diff
@@ -110,6 +114,201 @@ static const int NumThreadsPerQuadPair = NumThreadsPerQuad * 2;
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+CUTLASS_HOST_DEVICE uint ThreadIdxX() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::local_id::x();
+#else
+  return threadIdx.x;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint ThreadIdxY() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::local_id::y();
+#else
+  return threadIdx.y;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint ThreadIdxZ() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::local_id::z();
+#else
+  return threadIdx.z;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockIdxX() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_id::x();
+#else
+  return blockIdx.x;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockIdxY() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_id::y();
+#else
+  return blockIdx.y;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockIdxZ() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_id::z();
+#else
+  return blockIdx.z;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockDimX() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_range::x();
+#else
+  return blockDim.x;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockDimY() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_range::y();
+#else
+  return blockDim.y;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint BlockDimZ() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::work_group_range::z();
+#else
+  return blockDim.z;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint GridDimX() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::global_range::x();
+#else
+  return gridDim.x;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint GridDimY() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::global_range::y();
+#else
+  return gridDim.y;
+#endif
+}
+
+CUTLASS_HOST_DEVICE uint GridDimZ() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  return syclcompat::global_range::z();
+#else
+  return gridDim.z;
+#endif
+}
+
+// syncthreads
+
+CUTLASS_DEVICE void syncthreads() {
+#if defined(CUTLASS_ENABLE_SYCL)
+  syclcompat::wg_barrier();
+#else
+  __syncthreads();
+#endif
+}
+
+CUTLASS_DEVICE int syncthreads_and(int cond) {
+#if defined(CUTLASS_ENABLE_SYCL)
+  assert(false);
+#else
+  return __syncthreads_and(cond);
```

Comment:
```suggestion
#if !defined(CUTLASS_ENABLE_SYCL)
  return __syncthreads_and(cond);
```

Reply:
Added a todo comments. 

This is a WIP, It is better to just fail here that to continue and fail later because this function is not implemented properly.